src/test/java/io/pinecone/PineconeIndexOperationClientTest.java

```diff
@@ -53,12 +58,113 @@ public void testIndexCreation() throws IOException {
         when(mockResponseFuture.toCompletableFuture()).thenReturn(mockCompletableFuture);
 
         PineconeIndexOperationClient indexDeletionService = new PineconeIndexOperationClient(clientConfig, mockClient);
-        indexDeletionService.createIndex("testEnvironment", 128, "cosine");
+        CreateIndexRequest createIndexRequest = new CreateIndexRequest().indexName("test_name").dimension(3);
+        indexDeletionService.createIndex(createIndexRequest);
 
         verify(mockClient, times(1)).prepare(eq("POST"), anyString());
         verify(mockBoundRequestBuilder).setHeader(eq("accept"), eq("text/plain"));
         verify(mockBoundRequestBuilder).setHeader(eq("content-type"), eq("application/json"));
         verify(mockBoundRequestBuilder).execute();
         verify(mockClient).close();
     }
+
+    @Test
+    public void testIndexCreationThrowsExceptionWithNullIndex() {
+        AsyncHttpClient mockClient = mock(DefaultAsyncHttpClient.class);
+        PineconeClientConfig clientConfig = new PineconeClientConfig().withApiKey("testApiKey").withEnvironment("testEnvironment");
+        Response mockResponse = mock(Response.class);
+        BoundRequestBuilder mockBoundRequestBuilder = mock(BoundRequestBuilder.class);
+        ListenableFuture<Response> mockResponseFuture = mock(ListenableFuture.class);
+        CompletableFuture<Response> mockCompletableFuture = mock(CompletableFuture.class);
+
+        when(mockResponse.getStatusCode()).thenReturn(201);
+        when(mockClient.prepare(anyString(), anyString())).thenReturn(mockBoundRequestBuilder);
+        when(mockBoundRequestBuilder.setHeader(anyString(), anyString())).thenReturn(mockBoundRequestBuilder);
+        when(mockBoundRequestBuilder.setBody(anyString())).thenReturn(mockBoundRequestBuilder);
+        when(mockBoundRequestBuilder.execute()).thenReturn(mockResponseFuture);
+        when(mockResponseFuture.toCompletableFuture()).thenReturn(mockCompletableFuture);
+
+        PineconeIndexOperationClient indexDeletionService = new PineconeIndexOperationClient(clientConfig, mockClient);
```

Comment:
I think this variable name should not have "deletion" in it since the class does more things now. I'd say it's just the `indexService` now.

Reply:
Thank you for pointing it out! Will fix it.